gnomad/sample_qc/pipeline.py

```diff
@@ -401,111 +410,150 @@ def annotate_sex(
+            ploidy_ht = ploidy_ht.rename(
+                {
+                    "x_ploidy": "chrX_ploidy",
+                    "y_ploidy": "chrY_ploidy",
+                    "x_mean_dp": "chrX_mean_dp",
+                    "y_mean_dp": "chrY_mean_dp",
+                }
             )
-
-            # If the `variants_only_y_ploidy' is True modify the name of the normalization contig mean DP to indicate
-            # that this is the variant dataset only mean DP (this will have already been added if
-            # 'variants_only_x_ploidy' was also True).
-            if variants_only_y_ploidy:
-                ploidy_ht = ploidy_ht.annotate(
-                    **{
-                        f"var_data_{normalization_contig}_mean_dp": y_ploidy_idx.autosomal_mean_dp
-                    }
-                )
-
-        mt = mtds.variant_data
-    else:
-        mt = mtds
-        if is_sparse:
+        else:
             ploidy_ht = impute_sex_ploidy(
-                mt,
+                hl.filter_intervals(mt, ref_keep_locus_intervals),
                 excluded_intervals,
                 included_intervals,
                 normalization_contig,
-                use_only_variants=variants_only_x_ploidy,
+                use_only_variants=False,
             )
-            ploidy_ht = ploidy_ht.rename(
+        if variants_only_x_ploidy:
+            ploidy_ht = ploidy_ht.drop("chrX_ploidy", "chrX_mean_dp")
+        if variants_only_y_ploidy:
+            ploidy_ht = ploidy_ht.drop("chrY_ploidy", "chrY_mean_dp")
+
+    if var_keep_contigs:
+        logger.info(
+            "Filtering variants for variant only sex chromosome ploidy imputation."
+        )
+        filtered_mt = hl.filter_intervals(mt, var_keep_locus_intervals)
+        if variants_filter_lcr or variants_filter_segdup:
+            logger.info(
+                "Filtering out variants in %s",
+                f"segmental duplications {'and low confidence regions' if variants_filter_lcr else ''}"
+                if variants_filter_segdup
+                else " low confidence regions ",
+            )
+            filtered_mt = filter_low_conf_regions(
+                filtered_mt,
+                filter_lcr=variants_filter_lcr,
+                filter_decoy=False,
+                filter_segdup=variants_filter_segdup,
+            )
+        if variants_snv_only:
+            logger.info("Filtering to SNVs")
+            filtered_mt = filtered_mt.filter_rows(
+                hl.is_snp(filtered_mt.alleles[0], filtered_mt.alleles[1])
+            )
+
+        logger.info(
+            "Imputing sex chromosome ploidy using only variant depth information on the following contigs: %s",
+            var_keep_contigs,
+        )
+        if is_vds:
+            var_ploidy_ht = hl.vds.impute_sex_chromosome_ploidy(
+                hl.vds.VariantDataset(mtds.reference_data, filtered_mt),
+                calling_intervals=included_intervals,
```

Comment:
does this work if the default of None is used for calling_intervals?

Reply:
Good catch, nope, for now I will add a NotImplemented error